src/rdkafka_request.c

```diff
@@ -3343,17 +3343,15 @@ static int rd_kafka_handle_Produce_error(rd_kafka_broker_t *rkb,
                          * which should not be treated as a fatal error
                          * since this request and sub-sequent requests
                          * will be retried and thus return to order.
-                         * Unless the error was a timeout, or similar,
-                         * in which case the request might have made it
-                         * and the messages are considered possibly persisted:
-                         * in this case we allow the next in-flight response
-                         * to be successful, in which case we mark
-                         * this request's messages as succesfully delivered. */
-                        if (perr->status &
-                            RD_KAFKA_MSG_STATUS_POSSIBLY_PERSISTED)
-                                perr->update_next_ack = rd_true;
-                        else
-                                perr->update_next_ack = rd_false;
+                         * In case the message is possibly persisted
+                         * we still treat it as not persisted,
+                         * expecting DUPLICATE_SEQUENCE_NUMBER
+                         * in case it was persisted or NO_ERROR in case
+                         * it wasn't. Increasing next expected sequence
+                         * prematurely can lead to fatal errors in case
+                         * it wasn't persisted and the second one to retry
+                         * fails with OUT_OF_ORDER_SEQUENCE_NUMBER. */
+                        perr->update_next_ack = rd_false;
```

Comment:
I think we should remove the part from "Increasing next expected....". This paragraph is explaining the current issue that we are fixing not what next line of the code is doing. It is confusing the statement.